contracts/RNGChainlinkV2.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.6;
+
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/dev/VRFConsumerBaseV2.sol";
+import "@pooltogether/owner-manager-contracts/contracts/Manageable.sol";
+
+import "./interfaces/RNGChainlinkV2Interface.sol";
+
+contract RNGChainlinkV2 is RNGChainlinkV2Interface, VRFConsumerBaseV2, Manageable {
+  /* ============ Global Variables ============ */
+
+  /// @dev Reference to the VRFCoordinatorV2 deployed contract
+  VRFCoordinatorV2Interface internal _vrfCoordinator;
+
+  /// @dev A counter for the number of requests made used for request ids
+  uint32 public requestCounter;
+
+  /// @dev A list of random numbers from past requests mapped by request id
+  mapping(uint32 => uint256) internal randomNumbers;
+
+  /// @dev A list of blocks to be locked at based on past requests mapped by request id
+  mapping(uint32 => uint32) internal requestLockBlock;
+
+  /// @dev A mapping from Chainlink request ids to internal request ids
+  mapping(uint256 => uint32) internal chainlinkRequestIds;
+
+  /// @notice Chainlink VRF subscription request configuration
+  RequestConfig public sRequestConfig;
+
+  /// @notice Chainlink VRF subscription request id
+  uint256 public sRequestId;
```

Comment:
What is this?

Reply:
We store the most recent Chainlink VRF request id to compare it with the request id passed to the `fulfillRandomWords` fallback function.
If it doesn't match, the transaction will revert.
https://github.com/pooltogether/pooltogether-rng-contracts/blob/701badd75899625cd0a1f5bf3271b5c6aa5fa21e/contracts/RNGChainlinkV2.sol#L210